charts/external-dns/templates/deployment.yaml

```diff
@@ -132,6 +135,47 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+        {{- if eq $provider "webhook" }}
+        {{- with .Values.provider }}
+        - name: {{ $providername }}
+          image: {{ get .images .name }}:{{- .version }}
+          imagePullPolicy: {{ $.Values.image.pullPolicy }}
+          {{- with .env }}
+          env:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          {{- with .args }}
+          args:
+            - {{ toYaml . | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: http
+              protocol: TCP
+              containerPort: 8888
```

Comment:
It is problematic to expose port 8888 outside the pod. If you encourage providers to expose metrics through the same port they are exposing the webhook API on, that will encourage admins to open that port beyond localhost, allowing things outside the pod to make DNS changes as if they were external-dns.

Reply:
AFAIK, It's not exposing port. 

In [kubernetes doc](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#ports), it is written:
> Not specifying a port here DOES NOT prevent that port from being exposed.

I agree with you that we should do something on metrics, and encourage use of another port. 
It's not in current spec of webhook provider and it's not the purpose of this PR.